devito/stencilkernel.py

```diff
@@ -0,0 +1,131 @@
+from hashlib import sha1
+from itertools import chain
+from os import path
+
+import cgen as c
+import numpy as np
+
+from devito.compiler import (get_compiler_from_env, get_tmp_dir,
+                             jit_compile_and_load)
+from devito.expression import Expression
+from devito.interfaces import SymbolicData
+from devito.iteration import Iteration
+from devito.logger import error
+from devito.tools import filter_ordered
+
+__all__ = ['StencilKernel']
+
+
+class StencilKernel(object):
+    """Code generation class, alternative to Propagator"""
```

Comment:
Ultimately yes, but whether we retro-fit this type of internals into the current Propagator or whether we re-implement all the missing current features is up for discussion. This is really just intended as a proof-of-concept that we can be generic across dimensions.